### PR TITLE
[ISSUE#48] add https status code enum

### LINKED ISSUE #48: [Test support] add enum with status codes that are currently in use
### Challenge
We have identified a repetition in our code where variables are used to store the expected status codes for assertions.

### Solution
Since we decided not to use any external library that would include the list of status codes, having an enum with a list of status code currently used will come in handy.

### PR DESCRIPTION
**Pull request by bugITwhisperer**
### Scope of changes
We decided to introduce an enum that contains a list of the status codes currently in use. 
This will improve the overall project maintainability, and at the same time save us from code repetition.

- enums directory created in `src\api\enums`
- enum file created: `src\api\enums\api-status-code.enum.ts` with 5 status codes: 200, 400, 401, 404, 422

### How to use it in tests
- import file via:
```
import { HttpStatusCode } from "@_src_api/enums/api-status-code.enum";
```

-  when writing assertion use enum name and enum's member name, e.g.: 
```
expect(code).toBe(HttpStatusCode.Unauthorized401);
```

- when writing an assertion that uses status code that is not a member of the enum, please adjust the file and add it to the enum by following convention: `{name}{code} = {code}`
You can check the name using this [gist](https://gist.github.com/RWOverdijk/6cef816cfdf5722228e01cc05fd4b094) 

### PR steps completed:
- [x] The PR is associated with the relevant ISSUE
- [x] All new and existing tests have PASS status
- [x] Commit messages follow the `kawqa-gad-playwright` project's [git commit message format](https://github.com/kat-kan/kawqa-gad-playwright/CONTRIBUTING.md)

### Additional changes - not related to the initial PR
- Husky pre-commit: add feat and fix to branch validation regex `\.husky\pre-commit`

### CODE REVIEW

**Review by adamcegielka — DISMISSED** [review]

**Review by kat-kan — DISMISSED** [review]
Discussion topic: what would be our approach for adjusting existing tests to this change?

File: src/api/enums/api-status-code.enum.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export enum HttpStatusCode {
+    Ok200 = 200,
```
[the remark this answers is not in the document]
Comment (kat-kan) [reply src/api/enums/api-status-code.enum.ts:2]: @bugITwhisperer @adamcegielka @KarolinaZakrzewska still waiting for Your input ;)

[the remark this answers is not in the document]
Comment (kat-kan) [reply src/api/enums/api-status-code.enum.ts:2]: Also @Justyna-KO we invite You to join the discussion about the preferred option (https://github.com/kat-kan/kawqa-gad-playwright/pull/50#discussion_r1404590594)

[the remark this answers is not in the document]
Comment (Justyna-KO) [reply src/api/enums/api-status-code.enum.ts:2]: I believe that we will have no problem remembering the status codes. In my opinion, we can choose the first, shorter option.

[the remark this answers is not in the document]
Comment (bugITwhisperer) [reply src/api/enums/api-status-code.enum.ts]: @kat-kan , @Justyna-KO: Perfect, thank you for the feedback guys :) I've updated the file as per your choice. Could you please re-review the changes?

**Review by kat-kan — APPROVED** [review]

**Review by Justyna-KO — APPROVED** [review]

**Comment by bugITwhisperer** [comment]
@kat-kan: let's discuss our approach shortly during tomorrow's video meeting. I've already prepared the changes so if you agree they're fine, I can raise PR to address this

**Pull request merged**